DSpace/dspace-angular
# Browse clarify filter functionality

## References
- Fixes https://github.com/DSpace/dspace-angular/issues/1379
- Related to https://github.com/DSpace/DSpace/issues/8203

## Description
UX related changes for the browse to clarify the startsWith function is a filter.

## Instructions for Reviewers
- 'Go back' button "All browse results" that when filtering you can go back to all browse results without having to send empty query
  - Reset on both `startsWith` (so query to filter) and `value` (when a value like subject is selected and it shows items based on that value)
- Placeholder text search bar changed to indicate it's a filter
- Hover tooltip on next button indicating no more next pages of results available

The alphabet row was removed (see comment here: https://github.com/DSpace/DSpace/issues/8203#issuecomment-1064263899) :

- The "0-9" option doesn't work with filters, since it send `startsWith=0`
  - It could for example send `startsWith=[0 TO 9]` but this is not immediately supported REST side
    - Ex: [/discover/browses/subject/entries?sort=default,ASC&page=0&size=20&startsWith=5](https://api7.dspace.org/server/#/server/api/discover/browses/subject/entries?sort=default,ASC&page=0&size=20&startsWith=5) => has 1 result
      [startsWith=[0 TO 9]](https://api7.dspace.org/server/#/server/api/discover/browses/subject/entries?sort=default,ASC&page=0&size=20&startsWith=%5B0%20TO%209%5D) => No results
      [startsWith=[0||1||2||3||4||5||6||7||8||9]](https://api7.dspace.org/server/#/server/api/discover/browses/subject/entries?sort=default,ASC&page=0&size=20&startsWith=%5B0%7C5%5D) => No results
- The alphabet letters itself only support latin alphabet, could be changed to be dependent on configured list of 'letters' based on language selected, but was outside of scope for this issue

## Testing 

- Test [[original issue](https://github.com/DSpace/dspace-angular/issues/1379)](https://github.com/DSpace/dspace-angular/issues/1379) case => Should be more clear it is filter
- Go to value browse (eg subject) => Filter by typing in prefix, test the go back button
- Go to value browse (eg subject) => Go to list of items with that subject by pressing that subject, test the go back button

## Checklist
